api/client.go

```diff
@@ -84,6 +84,51 @@ func DefaultConfig() *Config {
 	return config
 }
 
+// ConfigureTLS takes the CA, client and TLS configuration and configures the
+// the HTTP client TLS config.
+func (c *Config) ConfigureTLS(CACert, CAPath, clientCert, clientKey,
```

Comment:
These values should be put into a struct, rather than an endless number of parameters.